marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/ConstraintsSpec.hs

```diff
@@ -0,0 +1,178 @@
+import qualified Language.Marlowe.Runtime.ChainSync.Api as Chain
+import qualified Language.Marlowe.Runtime.Core.Api as Core.Api
+import Language.Marlowe.Runtime.Transaction.Constraints (TxConstraints(..))
+import qualified Language.Marlowe.Runtime.Transaction.Constraints as Constraints
+import qualified Language.Marlowe.Runtime.Transaction.Constraints as TxConstraints
+import Test.Hspec (Spec, expectationFailure, shouldBe)
+import qualified Test.Hspec as Hspec
+import qualified Test.Hspec.QuickCheck as Hspec.QuickCheck
+import qualified Test.QuickCheck as QuickCheck
+import Test.QuickCheck.Hedgehog (hedgehog)
+
+byteStringGen :: QuickCheck.Gen ByteString
+byteStringGen = BS.pack <$> QuickCheck.arbitrary
+
+emptyTxConstraints :: TxConstraints 'Core.Api.V1
+emptyTxConstraints = TxConstraints
+  { marloweInputConstraints = TxConstraints.MarloweInputConstraintsNone
+  , payoutInputConstraints = Set.empty
+  , roleTokenConstraints = TxConstraints.SpendRoleTokens Set.empty
+  , payToAddresses = Map.empty
+  , payToRoles = Map.empty
+  , marloweOutputConstraints = TxConstraints.MarloweOutputConstraintsNone
+  , merkleizedContinuationsConstraints = mempty
+  , signatureConstraints = Set.empty
+  , metadataConstraints = Map.empty
+  }
+
+spec :: Spec
+spec = do
+  Hspec.QuickCheck.prop "mustSpendToken" do
+    policyId <- Chain.PolicyId <$> byteStringGen
+    tokenName <- Chain.TokenName <$> byteStringGen
+
+    let
+      assetId :: Chain.AssetId
+      assetId = Chain.AssetId policyId tokenName
+
+      actual :: Constraints.TxConstraints 'Core.Api.V1
+      actual = Constraints.mustSpendRoleToken assetId
+
+      expected :: Constraints.TxConstraints 'Core.Api.V1
+      expected = emptyTxConstraints
+        { roleTokenConstraints = Constraints.SpendRoleTokens $ Set.singleton assetId }
+
+    pure $ actual `shouldBe` expected
+
+  Hspec.describe "adjustTxForMinUtxo" do
+
+    Hspec.QuickCheck.prop "The Marlowe output has NOT been adjusted" do
+      protocol <- hedgehog genProtocolParameters
+
+      marloweScriptHash <- hedgehog genScriptHash
+      let
+        scriptAddress hash = fromCardanoAddressAny
+          $ C.AddressShelley
+          $ C.makeShelleyAddress C.Mainnet (C.PaymentCredentialByScript hash) C.NoStakeAddress
+        marloweAddress = scriptAddress marloweScriptHash
+
+        getValueAtAddress :: Chain.Address -> [C.TxOut C.CtxTx C.BabbageEra] -> Maybe (C.TxOutValue C.BabbageEra)
+        getValueAtAddress targetAddress txos = getFirst . mconcat . map (First
+          . (\(C.TxOut addressInEra txOutValue _ _) ->
+              if fromCardanoAddressInEra C.BabbageEra addressInEra == targetAddress
+                then Just txOutValue
+                else Nothing))
+          $ txos
+
+      txBodyContent <- hedgehog $ genTxBodyContent C.BabbageEra
+
+      let
+        actual = (getValueAtAddress marloweAddress . C.txOuts)
+          <$> Constraints.adjustTxForMinUtxo protocol marloweAddress txBodyContent
```

Comment:
We need to modify the `txBodyContent` so that it actually contains an output to the marlowe address here, or `getValueAtAddress` will always return `Nothing`, leading to a false positive test case.